File: CoreAutomata/IPlatformImpl.cs
using System;
using System.IO;

namespace CoreAutomata
{
    public interface IPlatformImpl
    {
        Region WindowRegion { get; }

        bool DebugMode { get; }

        void Toast(string Msg);

        IPattern LoadPattern(Stream Stream, bool Mask);

        IPattern GetResizableBlankPattern();

        void MessageBox(string Title, string Message);

        void Highlight(Region Region, TimeSpan Timeout);
    }
}

File: app/src/main/java/com/mathewsachin/fategrandautomata/core/IPlatformImpl.kt
package com.mathewsachin.fategrandautomata.core

import java.io.InputStream
import kotlin.time.Duration

interface IPlatformImpl {
    val windowRegion: Region

    fun toast(Message: String)

    fun loadPattern(Stream: InputStream): IPattern

    fun getResizableBlankPattern(): IPattern

    fun messageBox(Title: String, Message: String)

    fun highlight(Region: Region, Duration: Duration)
}